geps/gep-1619.md

```diff
@@ -62,14 +62,110 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
+implementation details.
+
+#### Gateway Initiated Sessions
+
+This represents the most straightforward scenario for utilizing cookies. When a request is made, the gateway includes
+the `set-cookie` header in the final response, prompting the client to generate a cookie. This cookie is subsequently
+used in future requests, allowing the gateway to consistently choose the same upstream, establishing a persistent
+session.
+
+Here an example implementation of this process:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B-->>-G: Response
+    G->>G: Add set-cookie header
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br> with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup of<br>server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Backend Initiated Sessions
+
+When dealing with backend-initiated sessions, the process becomes somewhat more complex. The gateway must modify the
+cookie's value to establish persistent sessions. However, in the most straightforward scenario, if the backend is not
+configured for session persistence, it should refrain from modifying the cookie. In this situation, the gateway should
+remain passive and simply forward the `set-cookie` header as it is.
+
+Here's an example of where the backend initiating a session, but the gateway does not modify the cookie:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B->>B: Add set-cookie<br>header
+    B-->>-G: Response<br>[set-cookie]
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br> with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    Note right of G: Gateway is not configured<br>for session persistence
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+Should an application developer wish for the gateway to establish a persistent session, even when a cookie with the same
+name is generated by the backend server, the gateway will need to modify the set-cookie header provided by the backend 
+server. The primary motivation behind gateways updating the cookie is to rewrite the cookie value with a server
+identifier in order to a maintain persistent session to a backend. There are several approaches to updating a cookie's
```

Comment:
I see this in several places - not sure if insisting on cookie rewriting is as an example, but in practice it's the more complex implementation. Adding a second cookie is simpler. So maybe clarify "several approaches to update a cookie value or add more cookies".